assets/src/app/services/workflow.service.ts

```diff
@@ -1,31 +1,45 @@
 
 import { Injectable } from '@angular/core'
+import { formatDate } from '@angular/common'
 import { HttpClient, HttpParams, HttpHeaders } from '@angular/common/http'
 import { Observable, of } from 'rxjs'
 import { catchError, map, tap } from 'rxjs/operators'
 
-import { WorkflowPage, WorkflowData, WorkflowHistory } from '../models/page/workflow_page'
+import { WorkflowQueryParams, WorkflowPage, WorkflowData, WorkflowHistory } from '../models/page/workflow_page'
 import { Step, Workflow, WorkflowEvent } from '../models/workflow'
 import { StartWorkflowDefinition } from '../models/startWorkflowDefinition'
 
 @Injectable()
 export class WorkflowService {
   private workflowUrl = '/api/workflow'
   private workflowsUrl = '/api/step_flow/workflows'
+  private workflowIdentifiersUrl = '/api/step_flow/definitions_identifiers'
   private workflowsLauncher = '/api/step_flow/launch_workflow'
   private workflowDefinitionsUrl = '/api/step_flow/definitions'
   private statisticsUrl = '/api/step_flow/workflows_statistics'
 
   constructor(private http: HttpClient) { }
 
-  getWorkflowDefinitions(page?: number, per_page?: number): Observable<WorkflowPage> {
+  getWorkflowDefinitions(page?: number, per_page?: number, right_action?: string, search?: string, versions?: string[], mode?: string): Observable<WorkflowPage> {
```

Comment:
move close to the getWorkflowDefinition, don't mix with getWorkflows

Reply:
can you explain what you mean please?

Reply:
in the code it's currently:
getWorkflowDefinitions
getWorkflows
getWorkflowDefinition

reorder with something more relevant like:
getWorkflowDefinitions
getWorkflowDefinition
getWorkflows